core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -333,7 +333,7 @@ public T acquire(long time, TimeUnit unit) throws TimeoutException, IOException
         }
         long currTimeMs = mClock.millis();
         try {
-          if (currTimeMs >= endTimeMs || !mNotEmpty
+          if (endTimeMs - currTimeMs <= 0 || !mNotEmpty
```

Comment:
can you put a comment here that we do the comparison in this way to prevent overflow?